cajita/src/Cajita_GlobalGrid_impl.hpp

```diff
@@ -211,6 +211,74 @@ GlobalGrid<MeshType>::blockRank( const int i, const int j ) const
     return blockRank( cr );
 }
 
+//---------------------------------------------------------------------------//
+// Get the MPI sub communicator for given direction
+template <class MeshType>
+MPI_Comm GlobalGrid<MeshType>::subComm( const int dim ) const
+{
+    MPI_Comm sub_comm;
+
+    // Set colors same within the dimBlockId of the dim
+    MPI_Comm_split( comm(), dimBlockId( dim ), blockId(), &sub_comm );
+    return sub_comm;
+}
+
+//---------------------------------------------------------------------------//
+// Get the MPI sub communicator for reduction in the given direction
```

Comment:
This creates a sub communicator in 2 dimensions, right?

Reply:
Yes. create sub communicator in 2D plane for 3D and sub communicator in 1D line for 2D.